src/tools/illink/src/linker/Linker.Steps/MarkStep.cs

```diff
@@ -1810,11 +1791,6 @@ bool ShouldWarnForReflectionAccessToCompilerGeneratedCode (FieldDefinition field
 
 		void ProcessAnalysisAnnotationsForField (FieldDefinition field, DependencyKind dependencyKind, in MessageOrigin origin)
 		{
-			if (origin.Provider != ScopeStack.CurrentScope.Origin.Provider) {
-				Debug.Assert (dependencyKind == DependencyKind.DynamicallyAccessedMemberOnType ||
-					(origin.Provider is MethodDefinition originMethod && CompilerGeneratedState.IsNestedFunctionOrStateMachineMember (originMethod)));
-			}
-
```

Comment:
This assertion was to make sure that we always passed ScopeStack.CurrentScope.Origin unless the call to MarkField originated from the ReflectionMarker. It always used the parameter `origin`, so the behavior is the same and this check isn't necessary.